From d66d5661a67b315a0abaf2b0f23e5a79afdc2501 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 2 Mar 2024 15:08:57 +0100
Subject: rtc: Print which error caused RTC read failure

Currently we can't know from kernel log, whether the read failed due
to I2C bus error, or invalid time value, etc. Print the error in error
message.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/rtc/class.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/class.c b/drivers/rtc/class.c
index e31fa0ad127e..2f14aecf0e24 100644
--- a/drivers/rtc/class.c
+++ b/drivers/rtc/class.c
@@ -66,7 +66,8 @@ static void rtc_hctosys(struct rtc_device *rtc)
 	err = rtc_read_time(rtc, &tm);
 	if (err) {
 		dev_err(rtc->dev.parent,
-			"hctosys: unable to read the hardware clock\n");
+			"hctosys: unable to read the hardware clock %pe\n",
+			ERR_PTR(err));
 		goto err_read;
 	}
 
-- 
2.35.3

